src/types/runtime.go

```diff
@@ -13,12 +13,13 @@ type ZarfCommonOptions struct {
 
 // ZarfDeployOptions tracks the user-defined preferences during a package deployment.
 type ZarfDeployOptions struct {
-	Insecure     bool              `json:"insecure" jsonschema:"description=Allow insecure connections for remote packages"`
-	Shasum       string            `json:"shasum" jsonschema:"description=The SHA256 checksum of the package to deploy"`
-	PackagePath  string            `json:"packagePath" jsonschema:"description=Location where a Zarf package to deploy can be found"`
-	Components   string            `json:"components" jsonschema:"description=Comma separated list of optional components to deploy"`
-	SGetKeyPath  string            `json:"sGetKeyPath" jsonschema:"description=Location where the public key component of a cosign key-pair can be found"`
-	SetVariables map[string]string `json:"setVariables" jsonschema:"description=Key-Value map of variable names and their corresponding values that will be used to template against the Zarf package being used"`
+	Insecure        bool              `json:"insecure" jsonschema:"description=Allow insecure connections for remote packages"`
+	Shasum          string            `json:"shasum" jsonschema:"description=The SHA256 checksum of the package to deploy"`
+	PackagePath     string            `json:"packagePath" jsonschema:"description=Location where a Zarf package to deploy can be found"`
+	Components      string            `json:"components" jsonschema:"description=Comma separated list of optional components to deploy"`
+	SGetKeyPath     string            `json:"sGetKeyPath" jsonschema:"description=Location where the public key component of a cosign key-pair can be found"`
+	SetVariables    map[string]string `json:"setVariables" jsonschema:"description=Key-Value map of --set variable names and their corresponding values that will be used to template against the Zarf package being used"`
```

Comment:
I probably need to chat about this one, want to understand the goal with this change.

Reply:
Short answer is that it enables you to do this:

`zarf-config.toml`

```
[package.create.set]
# The image tag used for the zarf agent, defaults to a dev image tag
agent_image = 'agent:local'

# Tag for the zarf injector binary to use
injector_version = 'pr-948-e699899'
```

```
$ zarf init --set AGENT_IMAGE=agent:v0.23.6 --confirm
```

By default the `--set` array is _either_ the one from the config _or_ the one set on the command line (which in the above example doesn't contain `INJECTOR_VERSION`).   This allows the two to be merged with `--set` on the CLI taking precedent over the config

Reply:
Refactored this out of pkg as requested.